pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -247,29 +247,41 @@ func getDockerRegistryLocations(lister listers.ServiceLister, location serviceLo
 		return []string{}
 	}
 
-	hasClusterIP := (len(service.Spec.ClusterIP) > 0) && (net.ParseIP(service.Spec.ClusterIP) != nil)
-	if hasClusterIP && len(service.Spec.Ports) > 0 {
-		svcPort := service.Spec.Ports[0].Port
-		ret := []string{
-			net.JoinHostPort(service.Spec.ClusterIP, fmt.Sprintf("%d", svcPort)),
-			net.JoinHostPort(fmt.Sprintf("%s.%s.svc", service.Name, service.Namespace), fmt.Sprintf("%d", svcPort)),
+	ip := net.ParseIP(service.Spec.ClusterIP)
+	if ip == nil {
+		return []string{}
+	}
+
+	if len(service.Spec.Ports) == 0 {
+		return []string{}
+	}
+
+	svcPort := service.Spec.Ports[0].Port
+	ret := []string{
+		net.JoinHostPort(fmt.Sprintf("%s.%s.svc", service.Name, service.Namespace), fmt.Sprintf("%d", svcPort)),
+	}
+
+	// Bug 1780376: add ClusterIP as a location if service supports IPv4
+	// IPv6 addresses are not valid locations in an image pull spec
+	ipv4 := ip.To4()
+	if ipv4 != nil {
+		ret = append(ret, net.JoinHostPort(ipv4.String(), fmt.Sprintf("%d", svcPort)))
+	}
```

Comment:
If the IP address is an IPv6 address, `ip.To4()` will return `nil`.

Note that IPv6 addresses that start with `::FFFF:` can be converted to IPv4.